x-pack/plugins/ml/public/application/components/import_export_jobs/export_jobs_flyout/export_jobs_flyout.tsx

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState, useEffect, useMemo } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiFlyout,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiFlyoutBody,
+  EuiTitle,
+  EuiSpacer,
+  EuiCheckbox,
+  EuiTabs,
+  EuiTab,
+} from '@elastic/eui';
+
+import { useMlApiContext, useMlKibana } from '../../../contexts/kibana';
+import { utilsProvider } from './utils';
+import { toastNotificationServiceProvider } from '../../../services/toast_notification_service';
+import type { JobType } from '../../../../../common/types/saved_objects';
+
+interface Props {
+  isDisabled: boolean;
+  currentTab: JobType;
+}
+
+export const ExportJobsFlyout: FC<Props> = ({ isDisabled, currentTab }) => {
+  const mlApiServices = useMlApiContext();
+  const {
+    getJobs,
+    dataFrameAnalytics: { getDataFrameAnalytics },
+  } = mlApiServices;
+
+  const {
+    services: {
+      notifications: { toasts },
+    },
+  } = useMlKibana();
+
+  const { exportAnomalyDetectionJobs, exportDataframeAnalyticsJobs } = utilsProvider(mlApiServices);
+
+  const [showFlyout, setShowFlyout] = useState(false);
+  const [adJobIds, setAdJobIds] = useState<string[]>([]);
+  const [dfaJobIds, setDfaJobIds] = useState<string[]>([]);
+  const [selectedJobIds, setSelectedJobIds] = useState<string[]>([]);
+  const [exporting, setExporting] = useState(false);
+  const [selectedJobType, setSelectedJobType] = useState<JobType>(currentTab);
+  const { displayErrorToast, displaySuccessToast } = useMemo(
+    () => toastNotificationServiceProvider(toasts),
+    [toasts]
+  );
+
+  useEffect(() => {
+    setAdJobIds([]);
+    setSelectedJobIds([]);
+    setExporting(false);
+    setSelectedJobType(currentTab);
+
+    if (showFlyout) {
+      getJobs().then(({ jobs }) => {
+        setAdJobIds(jobs.map((j) => j.job_id));
+      });
+      getDataFrameAnalytics().then(({ data_frame_analytics: dataFrameAnalytics }) => {
+        setDfaJobIds(dataFrameAnalytics.map((j) => j.id));
+      });
+    }
+  }, [showFlyout]);
+
+  function toggleFlyout() {
+    setShowFlyout(!showFlyout);
+  }
+
+  async function onExport() {
+    setExporting(true);
+    const title = i18n.translate('xpack.ml.importExport.exportFlyout.exportDownloading', {
+      defaultMessage: 'Your file is downloading in the background',
+      values: { count: selectedJobIds.length },
+    });
+    displaySuccessToast(title);
+
+    try {
+      if (selectedJobType === 'anomaly-detector') {
+        await exportAnomalyDetectionJobs(selectedJobIds);
+      } else {
+        await exportDataframeAnalyticsJobs(selectedJobIds);
+      }
```

Comment:
The file will only contain anomaly detection jobs or data frame analytics jobs.
Missing both in the same file would add extra complications with importing plus we'd need some meta data to identify which jobs are which type etc.

Reply:
makes sense, but we can just generate two files with one click, can't we?

Reply:
yes I believe we could generate two files.
This could be a future improvement, if we want to allow the user to make the selections for both types of jobs and then export them all out.